shared/build.gradle.kts

```
@@ -93,3 +120,61 @@ android {
 tasks.check {
     dependsOn(detekt)
 }
+
+buildkonfig {
+    packageName = "co.nimblehq.blisskmmic"
+
+    defaultConfigs {
+        buildConfigField(
+            STRING,
+            "CLIENT_ID",
+            BuildKonfig.CLIENT_ID
+        )
+        buildConfigField(
+            STRING,
+            "CLIENT_SECRET",
+            BuildKonfig.CLIENT_SECRET
+        )
+        buildConfigField(
+            STRING,
+            "BASE_URL",
+            BuildKonfig.STAGING_BASE_URL
+        )
+    }
+
+    defaultConfigs("production") {
+        buildConfigField(
+            STRING,
+            "CLIENT_ID",
+            BuildKonfig.CLIENT_ID
+        )
+        buildConfigField(
+            STRING,
+            "CLIENT_SECRET",
+            BuildKonfig.CLIENT_SECRET
+        )
+        buildConfigField(
+            STRING,
+            "BASE_URL",
+            BuildKonfig.PRODUCTION_BASE_URL
+        )
+    }
+
+    defaultConfigs("staging") {
+        buildConfigField(
+            STRING,
+            "CLIENT_ID_STAGING",
+            BuildKonfig.CLIENT_ID
+        )
+        buildConfigField(
+            STRING,
+            "CLIENT_SECRET_STAGING",
+            BuildKonfig.CLIENT_SECRET
+        )
```

We're defining different client id + secret while pointing to the same id + secret in BuildKonfig? Should they be different between env staging and production?

I think something went wrong with the buildconfig field names here. The field name should be the same between env 💭 

```suggestion
        buildConfigField(
            STRING,
            "CLIENT_ID",
            BuildKonfig.CLIENT_ID
        )
        buildConfigField(
            STRING,
            "CLIENT_SECRET",
            BuildKonfig.CLIENT_SECRET
        )
        buildConfigField(
            STRING,
            "BASE_URL",
            BuildKonfig.PRODUCTION_BASE_URL
        )
    }

    defaultConfigs("staging") {
        buildConfigField(
            STRING,
            "CLIENT_ID",
            BuildKonfig.CLIENT_ID_STAGING
        )
        buildConfigField(
            STRING,
            "CLIENT_SECRET",
            BuildKonfig.CLIENT_SECRET_STAGING
        )
```

---

Nice catch, thank you.